go-controller/pkg/config/config.go

```diff
@@ -399,6 +413,13 @@ type GatewayConfig struct {
 	V4JoinSubnet string `gcfg:"v4-join-subnet"`
 	// V6JoinSubnet to be used in the cluster
 	V6JoinSubnet string `gcfg:"v6-join-subnet"`
+	// V4MasqueradeSubnet to be used in the cluster
+	V4MasqueradeSubnet string `gcfg:"v4-masquerade-subnet"`
+	// V6MasqueradeSubnet to be used in the cluster
+	V6MasqueradeSubnet string `gcfg:"v6-masquerade-subnet"`
+	// masqueradeIps to be allocated from the masquerade subnets to enable host to service traffic
+	MasqueradeIPs MasqueradeIP
```

Comment:
This looks awkward. I would change it to be: MasqueradeIPs MasqueradeIPsConfig or something with a name that keeps them both plural